syntax = "proto3";

message PrimitiveObject {
  optional int64 int1 = 1;
  optional double float1 = 2;
  optional int64 int2 = 3;
  optional double float2 = 4;
}

message RepeatedObject {
  optional string str1 = 1;
  optional string str2 = 2;
  repeated int64 arr1 = 3;
  repeated string arr2 = 4;
}

message DictObject {
  map<string, int64> dict1 = 1;
  map<string, int64> dict2 = 2;
  map<string, string> dict3 = 3;
}

message CompositeObject {
  optional int64 int1 = 1;
  optional double float1 = 2;
  optional string str1 = 3;
  repeated int64 arr1 = 4;
  map<string, string> dict1 = 5;
}